### PR TITLE
EDU-178 Add cookie consent drawer

### PR DESCRIPTION
**Pull request by laurentiu-ilici**
Closes: https://educandu.atlassian.net/browse/EDU-178

### CODE REVIEW

File: src/components/cookie-consent-drawer.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+import { useTranslation } from 'react-i18next';
+
+const EDUCANDU_COOKIE_CONSENT = 'educanduCookieConsent';
+
+export default function CookieConsentDrawer() {
+  const consentCookie = cookie.get(EDUCANDU_COOKIE_CONSENT);
+  const [isVisible, setIsVisible] = useState(!consentCookie);
+
+  const handleClose = () => {
+    cookie.set(EDUCANDU_COOKIE_CONSENT, 'true', { expires: 365 });
```
Comment (laurentiu-ilici) [inline src/components/cookie-consent-drawer.js:13]: I don't know when to make it expire, I feel like either on logout or when you close the browser tab (if you aren't logged in).

File: src/components/cookie-consent-drawer.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    setIsVisible(false);
+  };
+
+  const { t } = useTranslation('cookieConsentDrawer');
+
+  return (
+    <Drawer title={t('consentTitle')} placement="bottom" visible={isVisible} closable={false}>
```
Comment (laurentiu-ilici) [inline src/components/cookie-consent-drawer.js:20]: I have some trouble with this drawer, it doesn't work like in the documentation, I'm curious if it work better on your machines, please let me know when you have a couple of minutes.

File: src/components/cookie-consent-drawer.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
+import { Button, Drawer } from 'antd';
+import React, { useState } from 'react';
+import cookie from '../common/cookie.js';
+import { useTranslation } from 'react-i18next';
+
+const EDUCANDU_COOKIE_CONSENT = 'educanduCookieConsent';
```
Comment (laurentiu-ilici) [inline src/components/cookie-consent-drawer.js:6]: Maybe this should be configurable as well

File: src/components/cookie-consent-drawer.js (hunk 1 above)
Comment (ahelmberger) [reply src/components/cookie-consent-drawer.js:13]: I would get pretty mad if I would see it everytime I open the browser and go to the website, so I like the 365 :-)

File: src/components/cookie-consent-drawer.js (hunk 3 above)
Comment (ahelmberger) [reply src/components/cookie-consent-drawer.js:6]: Maybe, I don't find it that important. It should maybe align a little bit better with the existing cookie names `SID` and `UILANG` (maybe `COOKIES`?)

File: src/components/cookie-consent-drawer.js (hunk 2 above)
Comment (ahelmberger) [reply src/components/cookie-consent-drawer.js:20]: Looks good to me (although you didn't really specify what was not working on your machine) :thinking:

Comment (laurentiu-ilici) [reply src/components/cookie-consent-drawer.js:19]: That's because several things didn't work

File: src/components/cookie-consent-drawer.js (hunk 3 above)
Comment (ahelmberger) [reply src/components/cookie-consent-drawer.js]: Cookies are sandboxed to the domain name anyway, so they wouldn't interfere, but anyway, it cannot hurt

File: src/components/cookie-consent-drawer.js (hunk 1 above)
Comment (laurentiu-ilici) [reply src/components/cookie-consent-drawer.js]: I've let it be

File: src/components/cookie-consent-drawer.js (hunk 2 above)
Comment (laurentiu-ilici) [reply src/components/cookie-consent-drawer.js:23]: Solved, thanks!

File: src/components/cookie-consent-drawer.js (hunk 3 above)
Comment (laurentiu-ilici) [reply src/components/cookie-consent-drawer.js]: I've made it an optional thing.

File: src/components/default-page-template.js [hunk 4]
```diff
@@ -172,6 +172,7 @@ function DefaultPageTemplate({ children, fullScreen, headerActions, alerts }) {
           ))}
         </div>
       </footer>
+      <CookieConsentDrawer cookieName={clientConfig.consentCookieName} />
```
Comment (ahelmberger) [inline src/components/default-page-template.js:175]: There's no need to pass down the client config as a prop. The `CookieConsentDrawer` has access to the client config itself.

File: src/resources/resources.json [hunk 5]
```diff
@@ -45,6 +45,22 @@
...
+      "consentTitle": "Before you start",
+      "consentText": "This website uses cookies that are technically needed for strictly functional aspects of the website. These cookies neither track your activities, nor provide 3-rd parties with information of any kind about your visit. By clicking accept you acknowledge this and give your express consent to the usage of the cookies."
+    }
+  },
+  {
+    "namespace": "cookieConsentDrawer",
+    "language": "de",
+    "resources": {
+      "consentTitle": "Bevor Sie fortfahren",
+      "consentText": "Diese Website verwendet technisch notwendige Cookies, um die Funktionalität der Website zu gewährleisten. Diese Cookies verfolgen weder Ihre Aktivitäten, noch liefern sie Dritten irgendwelche Informationen über Ihren Besuch. Indem Sie auf Akzeptieren klicken, bestätigen Sie dies und stimmen der Verwendung der Cookies ausdrücklich zu."
```
Comment (ahelmberger) [inline src/resources/resources.json:61]: Sounds good to me :+1:

File: src/components/default-page-template.js (hunk 4 above)
Comment (laurentiu-ilici) [reply src/components/default-page-template.js:175]: Hmm, how?

Comment (ahelmberger) [reply src/components/default-page-template.js:175]: `useService(ClientConfig)`?

Comment (laurentiu-ilici) [reply src/components/default-page-template.js]: Fixed.

**Review by ahelmberger — APPROVED** [review]

**Pull request merged**